openedx/core/djangoapps/programs/tests/test_views.py

```diff
@@ -0,0 +1,135 @@
+                self.user,
+                ['edX/DemoX_1/Run_1', 'edX/DemoX_2/Run_2', 'valid/edX/Course']
+            )
+            expected_output = {
+                'edX/DemoX_1/Run_1': {
+                    'category': 'xseries',
+                    'status': 'active',
+                    'course_codes': [
+                        {
+                            'organization': {'display_name': 'Test Organization', 'key': 'edX'},
+                            'marketing_slug': 'fake-marketing-slug',
+                            'display_name': 'Demo Course',
+                            'key': 'TEST_A',
+                            'run_modes': [
+                                {'sku': '', 'mode_slug': 'ABC_1', 'course_key': 'edX/DemoX_1/Run_1'},
+                                {'sku': '', 'mode_slug': 'ABC_2', 'course_key': 'edX/DemoX_2/Run_2'},
+                            ]
+                        }
+                    ],
+                    'subtitle': 'Dummy program for testing',
+                    'name': 'First Program'
+                },
+                'edX/DemoX_2/Run_2': {
+                    'category': 'xseries',
+                    'status': 'active',
+                    'course_codes': [
+                        {
+                            'organization': {'display_name': 'Test Organization', 'key': 'edX'},
+                            'marketing_slug': 'fake-marketing-slug',
+                            'display_name': 'Demo Course',
+                            'key': 'TEST_A',
+                            'run_modes': [
+                                {'sku': '', 'mode_slug': 'ABC_1', 'course_key': 'edX/DemoX_1/Run_1'},
+                                {'sku': '', 'mode_slug': 'ABC_2', 'course_key': 'edX/DemoX_2/Run_2'},
+                            ]
+                        }
+                    ],
+                    'subtitle': 'Dummy program for testing',
+                    'name': 'First Program'
+                },
+            }
+            self.assertEqual(expected_output, programs)
+
+    def test_get_course_programs_with_non_existing_courses(self):
+        """ Test that the method 'get_course_programs_for_dashboard' returns
+        only those program courses which exists in the programs api response.
+        """
+        # mock the request call
+        with patch('slumber.Resource.get') as mock_get:
+            mock_get.return_value = self.programs_api_response
+            self.assertEqual(
+                get_course_programs_for_dashboard(self.user, ['invalid/edX/Course']), {}
+            )
+
+    def test_get_course_programs_with_invalid_user(self):
+        """ Test that the method 'get_course_programs_for_dashboard' returns
+        empty dict for an anonymous user.
+        """
+        # mock the request call
+        with patch('slumber.Resource.get') as mock_get:
+            mock_get.return_value = self.programs_api_response
```

Comment:
as discussed earlier, instead of setting return_value, set side_effect to Exception, since a JSON response to an anonymous user is not a realistic scenario.